vars/deploy.groovy

```diff
@@ -0,0 +1,159 @@
+                        juStabUtils: utils,
+                        jenkinsUtilsStub: jenkinsUtils,
+                        deployType: deploymentType,
+                        cloudFoundry: target,
+                        mtaPath: script.commonPipelineEnvironment.mtarFilePath,
+                        deployTool: deployTool
+                    )
+
+                    utils.stash(stageName)
+                }
+                deployments["Deployment ${index > 1 ? index : ''}"] = {
+                    if (env.POD_NAME) {
+                        dockerExecuteOnKubernetes(script: script, containerMap: ContainerMap.instance.getMap().get(stageName) ?: [:]) {
+                            deployment.run()
+                        }
+                    } else {
+                        node(env.NODE_NAME) {
+                            deployment.run()
+                        }
+                    }
+                }
+                index++
+            }
+            utils.runClosures(deployments)
+        }
+        if (config.neoTargets) {
+            def source
+            if (config.source) {
+                source = config.source
+            } else {
+                def pom = readMavenPom file: 'application/pom.xml'
+                source = "application/target/${pom.getArtifactId()}.${pom.getPackaging()}"
+            }
+
+            for (int i = 0; i < config.neoTargets.size(); i++) {
+                def target = config.neoTargets[i]
+
+                Closure deployment = {
+                    utils.unstash(stageName)
+
+                    DeploymentType deploymentType
+                    if(enableZeroDowntimeDeployment) {
+                        deploymentType = DeploymentType.NEO_ROLLING_UPDATE
+                    }
+                    else {
+                        deploymentType = DeploymentType.selectFor(CloudPlatform.NEO, config.isProduction.asBoolean())
+                    }
+
+                    neoDeploy (
+                        script: script,
+                        warAction: deploymentType.toString(),
+                        source: source,
+                        neo: target
+                    )
+
+                    utils.stash(stageName)
+                }
+                deployments["Deployment ${index > 1 ? index : ''}"] = {
+                    if (env.POD_NAME) {
+                        dockerExecuteOnKubernetes(script: script, containerMap: ContainerMap.instance.getMap().get(stageName) ?: [:]) {
+                            deployment.run()
+                        }
+                    } else {
+                        node(env.NODE_NAME) {
+                            deployment.run()
+                        }
+                    }
+                }
+                index++
+            }
+            utils.runClosures(deployments)
+        }
+        if (!config.cfTargets && !config.neoTargets)
+        {
+            currentBuild.result = 'FAILURE'

```

Comment:
`currentBuild.result = 'FAILURE'` should not be needed as `error` does this implicitly

Reply:
Yes! I did remove it, but I did a reset and I did forget it. Thanks!